### PR TITLE
1.1.0 Metadata functions for Microsoft SQL Server

### PR DESCRIPTION
**Pull request by splatteredbits**

### CODE REVIEW

File: Tests/Test-YMsSqlExtendedProperty.Tests.ps1
Comment (decoyjoe) [inline Tests/Test-YMsSqlExtendedProperty.Tests.ps1:1]: Empty file?

File: Yodel/Functions/Invoke-YDbCommand.ps1 [hunk 1]
```diff
@@ -67,45 +82,57 @@ function Invoke-YDbCommand
...
-        [hashtable]$Parameter,
-        
+        [hashtable] $Parameter,
+
         [Parameter(Mandatory,ParameterSetName='ExecuteScalar')]
-        # Return the result as a single value instead of a row.  If the command returns multiple rows/columns, the value of the first row's first column is returned.
-        [switch]$AsScalar,
-        
+        # Return the result as a single value instead of a row.  If the command returns multiple rows/columns, the value
+        # of the first row's first column is returned.
```
Comment (decoyjoe) [inline Yodel/Functions/Invoke-YDbCommand.ps1:109]: Nit: Parameter docs should be above the `[Parameter()]` attribute.

File: Yodel/Functions/MsSql/Initialize-YMsSqlDatabase.ps1 [hunk 2]
```diff
@@ -0,0 +1,43 @@
+
+function Initialize-YMsSqlDatabase
+{
+    <#
+    .SYNOPSIS
+    Esnures a Microsoft SQL Server database exists.
```
Comment (decoyjoe) [inline Yodel/Functions/MsSql/Initialize-YMsSqlDatabase.ps1]: Typo: `Esnures`

File: Yodel/Functions/MsSql/Initialize-YMsSqlDatabase.ps1 [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+function Initialize-YMsSqlDatabase
+{
+    <#
+    .SYNOPSIS
+    Esnures a Microsoft SQL Server database exists.
+
+    .DESCRIPTION
+    The `Initialize-YMsSqlDatabase` function creates a database in Microsoft SQL Server if it doesn't already exist.
+    Pass the connection to SQL Server to the `Connection` parameter and the name of the database to the `Name`
+    parameter. If the database doesn't exist (i.e. there's no record for it in `sys.databases`), it is created.
```
Comment (decoyjoe) [inline Yodel/Functions/MsSql/Initialize-YMsSqlDatabase.ps1]: From these docs, I can't figure out how I would know if the function actually did anything or not. E.g. is anything returned? A message written?

**Review by decoyjoe — DISMISSED** [review]

**Comment by splatteredbits** [comment]
Came by to add another metadata function here and discovered this PR that had never been merged. Oops. 

Added two new functions `Get-YMsSqlTableColumn` and `Test-YMsSqlTableColumn`. Added `-PassThru` switch to all the `Test-*` functions. Added some missing tests.

@decoyjoe Ready for another review.

File: Yodel/Functions/MsSql/Get-YMsSqlTableColumn.ps1 [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        [Data.Common.DbConnection] $Connection,
+
+        # The table's schema. Defaults to `dbo`.
+        [String] $SchemaName = 'dbo',
+
+        # The table name.
+        [Parameter(Mandatory)]
+        [String] $TableName,
+
+        [String] $Name
```
Comment (decoyjoe) [inline Yodel/Functions/MsSql/Get-YMsSqlTableColumn.ps1:42]: Missing parameter doc.

File: Yodel/Functions/MsSql/Test-YMsSqlExtendedProperty.ps1 [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+        # The schema name whose extended property to test, or the schema name of the table whose extended property to
+        # test.
+        [Parameter(Mandatory, ParameterSetName='ForSchema')]
+        [Parameter(ParameterSetName='ForTable')]
+        [Parameter(ParameterSetName='ForTableColumn')]
+        [String] $SchemaName,
+
+        # The table name whose extended property to set, or the table name of the column whose extended property to
+        # set.
```
Comment (decoyjoe) [inline Yodel/Functions/MsSql/Test-YMsSqlExtendedProperty.ps1]: Should be "test" not "set".

File: Yodel/Functions/MsSql/Test-YMsSqlExtendedProperty.ps1 [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+        [Parameter(Mandatory, ParameterSetName='ForTable')]
+        [Parameter(Mandatory, ParameterSetName='ForTableColumn')]
+        [String] $TableName,
+
+        # The column name whose extended property to set.
```
Comment (decoyjoe) [inline Yodel/Functions/MsSql/Test-YMsSqlExtendedProperty.ps1]: Should be "test" not "set".

File: Yodel/Functions/MsSql/Test-YMsSqlTable.ps1 [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    Demonstrates how to test if a table exists by passing the table name to the `Name` parameter.
+    #>    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory)]
+        [Data.Common.DbConnection] $Connection,
+
+        [String] $SchemaName = 'dbo',
+
+        [Parameter(Mandatory)]
+        [String] $Name,
```
Comment (decoyjoe) [inline Yodel/Functions/MsSql/Test-YMsSqlTable.ps1]: Missing parameter docs.

File: Yodel/Functions/MsSql/Test-YMsSqlTableColumn.ps1 [hunk 8]
```diff
@@ -0,0 +1,58 @@
+
+function Test-YMsSqlTableColumn
+{
+    <#
+    .SYNOPSIS
+    Checks if a column exists on a table in Microsoft SQL Server.
+
+    .DESCRIPTION
+    The `Test-YMsSqlTableColumn` function tests if a column exists on a table in SQL Server. Pass the connection to SQL
+    Server to the `Connection` parameter, the table name to the `TabkeName` parameter, and the column name to the
```
Comment (decoyjoe) [inline Yodel/Functions/MsSql/Test-YMsSqlTableColumn.ps1]: Typo on `TabkeName`

**Review by decoyjoe — DISMISSED** [review]

**Review by decoyjoe — APPROVED** [review]

**Pull request merged**